### ISSUE TITLE
Opening project throws IllegalStateException

### ISSUE THREAD

**Issue by nikoliazekter** [UnderwaterApps/overlap2d#288]
When I hit "Open Project", Overlap2D immediately closes and throws ```java.lang.IllegalStateException: Provided directory does not exist!```.

**Comment by azakhary** [comment]
What's the OS?

**Comment by azakhary** [comment]
This is something new I guess. maybe there is a new bug when it ties to create the "default directory" and  it fails for some reason.

**Comment by nikoliazekter** [comment]
Windows 10. It happens in both github version (or whatever you call it) and downloaded from the official site. It doesn't even let me choose any directory.

**Comment by nikoliazekter** [comment]
Full console log:
```java
VisUI: Warning, using default favorites preference name for file chooser! (see FileChooser.setFavoritesPrefsName(String))
VisUI: Warning, using default favorites preference name for file chooser! (see FileChooser.setFavoritesPrefsName(String))
VisUI: Warning, using default favorites preference name for file chooser! (see FileChooser.setFavoritesPrefsName(String))
java.lang.IllegalStateException: Provided directory does not exist!
	at com.kotcrab.vis.ui.widget.file.FileChooser.setDirectory(FileChooser.java:653)
	at com.kotcrab.vis.ui.widget.file.FileChooser.setDirectory(FileChooser.java:649)
	at com.uwsoft.editor.view.menu.Overlap2DMenuBarMediator.showOpenProject(Overlap2DMenuBarMediator.java:229)
	at com.uwsoft.editor.view.menu.Overlap2DMenuBarMediator.handleFileMenuNotification(Overlap2DMenuBarMediator.java:148)
	at com.uwsoft.editor.view.menu.Overlap2DMenuBarMediator.handleNotification(Overlap2DMenuBarMediator.java:101)
	at com.puremvc.core.CoreView$$Lambda$21/229580448.onNotification(Unknown Source)
	at com.puremvc.patterns.observer.BaseObserver.notifyObserver(BaseObserver.java:82)
	at com.puremvc.core.CoreView.notifyObservers(CoreView.java:128)
	at com.puremvc.patterns.facade.SimpleFacade.notifyObservers(SimpleFacade.java:361)
	at com.puremvc.patterns.facade.SimpleFacade.sendNotification(SimpleFacade.java:323)
	at com.uwsoft.editor.event.MenuItemListener.changed(MenuItemListener.java:44)
	at com.badlogic.gdx.scenes.scene2d.utils.ChangeListener.handle(ChangeListener.java:28)
	at com.badlogic.gdx.scenes.scene2d.Actor.notify(Actor.java:181)
	at com.badlogic.gdx.scenes.scene2d.Actor.fire(Actor.java:146)
	at com.badlogic.gdx.scenes.scene2d.ui.Button.setChecked(Button.java:123)
	at com.badlogic.gdx.scenes.scene2d.ui.Button$1.clicked(Button.java:91)
	at com.badlogic.gdx.scenes.scene2d.utils.ClickListener.touchUp(ClickListener.java:89)
	at com.badlogic.gdx.scenes.scene2d.InputListener.handle(InputListener.java:58)
	at com.badlogic.gdx.scenes.scene2d.Stage.touchUp(Stage.java:348)
	at com.badlogic.gdx.InputMultiplexer.touchUp(InputMultiplexer.java:96)
	at com.badlogic.gdx.backends.lwjgl.LwjglInput.processEvents(LwjglInput.java:306)
	at com.badlogic.gdx.backends.lwjgl.LwjglCanvas$3.run(LwjglCanvas.java:234)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
Exception in thread "AWT-EventQueue-0" java.lang.RuntimeException: No OpenGL context found in the current thread.
	at org.lwjgl.opengl.GLContext.getCapabilities(GLContext.java:124)
	at org.lwjgl.opengl.GL11.glDeleteTextures(GL11.java:732)
	at com.badlogic.gdx.backends.lwjgl.LwjglGL20.glDeleteTexture(LwjglGL20.java:248)
	at com.badlogic.gdx.graphics.GLTexture.delete(GLTexture.java:170)
	at com.badlogic.gdx.graphics.Texture.dispose(Texture.java:194)
	at com.badlogic.gdx.graphics.g2d.TextureAtlas.dispose(TextureAtlas.java:418)
	at com.badlogic.gdx.scenes.scene2d.ui.Skin.dispose(Skin.java:389)
	at com.kotcrab.vis.ui.VisUI.dispose(VisUI.java:65)
	at com.uwsoft.editor.Overlap2D.dispose(Overlap2D.java:68)
	at com.badlogic.gdx.backends.lwjgl.LwjglCanvas$4.run(LwjglCanvas.java:309)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:726)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)

Process finished with exit code 0
```

**Comment by azakhary** [comment]
When you first run the editor, it normally should create a directory in your /Users/[yourusername/Documents/Overlap2D
I assume you do not have it as it failed to create it. (Can you please check)

If so, fast fix for you will be to create it by hand.

But ultimately it's a big blocker bug that has to be fixed urgently.

**Comment by azakhary** [comment]
It happens in ProjectManager line 91 initWorkspace() weird it does not happen. I have to reproduce on my machine first... but that's strange.

**Comment by nikoliazekter** [comment]
I have an em[ty Overlap2D folder.

**Comment by azakhary** [comment]
hmmmm

**Comment by azakhary** [comment]
I can't reproduce. are you running from sources? if so, is there any way you can run it in debug mode and let me know if Overlap2DMenuBarMediator.java:229 what is the return value of getDefaultWorkspace ? (and if it's something wicked - why? )

**Comment by nikoliazekter** [comment]
Yes, I'm running from sources. You probably meant getWorkspacePath(). It's empty. This ```editorConfigVO.lastOpenedSystemPath != null``` is true while it shouldn't be. And that's the cause of the problem. That's because ```lastOpenedSystemPath``` is initialized so
```public String lastOpenedSystemPath = "";``` 
So there are basically 2 options to fix this bug:
1. Don't set ```lastOpenedSystemPath``` to "", let it be null.
2. Instead of null check use ```lastOpenedSystemPath.length == 0```.

**Comment by azakhary** [comment]
Got it. Let's go with option 2. making default null, can cause other errors elsewhere.
I'll fix it tomorrow, but if you want to make pull request today, I'll merge it!

Thanks for finding this!

**Comment by azakhary** [comment]
Jenkins build scheduled. It will be on website in 5 minutes.

**Comment by azakhary** [comment]
Fixed in #290
